testsuites/syncgateway/upgrade/test_upgrade.py

```diff
@@ -0,0 +1,480 @@
+import os
+import random
+import time
+
+from keywords.couchbaseserver import verify_server_version
+from libraries.testkit.cluster import Cluster
+from keywords.utils import log_info, host_for_url
+from keywords.SyncGateway import (verify_sg_accel_version,
+                                  verify_sync_gateway_version,
+                                  verify_sg_accel_product_info,
+                                  verify_sync_gateway_product_info,
+                                  SyncGateway)
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.MobileRestClient import MobileRestClient
+from keywords import attachment
+from couchbase.bucket import Bucket
+from keywords.constants import SDK_TIMEOUT
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from requests.exceptions import HTTPError
+
+
+def test_upgrade(params_from_base_test_setup):
+    """ The initial versions of SG and CBS has already been provisioned at this point
```

Comment:
can you add @summary as we thought to start having that now.

Reply:
Done